### PR TITLE
refactor: require FoK params for DCA channels

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

**Review by GabrielBuragev — APPROVED** [review]

File: packages/shared/src/schemas.ts [hunk 1]
```diff
@@ -106,6 +120,8 @@ export const openSwapDepositChannelSchema = z
     fillOrKillParams: fillOrKillParams.optional(),
     dcaParams: dcaParams.optional(),
   })
+  .superRefine(ensureDcaWithFok)
+  // openSwapDepositChannelSchema
```
Comment (zoheb391) [inline packages/shared/src/schemas.ts:124]: ?

Comment (acdibble) [inline packages/shared/src/schemas.ts:124]:
```suggestion
```

File: packages/shared/src/__tests__/schema.test.ts [hunk 2]
```diff
@@ -78,17 +78,56 @@ describe('postSwapSchema', () => {
...
+        fillOrKillParams: {
+          retryDurationBlocks: 10,
+          refundAddress: '0x1234',
+          minPriceX128: '1',
+        },
+      }),
+    ).not.toThrow();
+  });
+
+  it('only allows FoK params without DCA params', () => {
```
Comment (zoheb391) [inline packages/shared/src/__tests__/schema.test.ts:121]:
```suggestion
  it('allows FoK params without DCA params', () => {
```

**Review by zoheb391 — APPROVED** [review]

**Comment by niklasnatter** [comment]
> Is this a protocol level requirement or only enforced by the SDK?

It is enforced by the SDK for now, but it might be added on the protocol level in the future.

**Comment by CumpsD** [comment]
> It is enforced by the SDK for now, but it might be added on the protocol level in the future.

Thanks!

**Pull request merged**